src/components/ui/DatePicker.tsx

```diff
@@ -0,0 +1,45 @@
+'use client';
+
+import { CalendarIcon } from '@/assets/images/icons';
+import {
+  Button,
+  Calendar,
+  Icon,
+  inputVariants,
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@/components/ui';
+import { cn } from '@/lib/tailwind/utils';
+
+import { format } from 'date-fns';
+import { SelectSingleEventHandler } from 'react-day-picker';
+
+interface IDatePickerProps {
+  date?: Date;
+  onSelect?: SelectSingleEventHandler;
+}
+
+export const DatePicker = ({ date, onSelect }: IDatePickerProps) => {
+  return (
+    <Popover>
+      <PopoverTrigger asChild>
+        <Button
+          variant="none"
+          size="none"
+          className={cn(
+            inputVariants(),
+            'flex items-center pr-4',
+            'data-[state=open]:ring-2 data-[state=open]:ring-primary-dark data-[state=open]:border-transparent',
```

Comment:
How about adding `whitespace-nowrap` css to avoid this screen collapse?
It might be tribial though.

<img width="178" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/25422441/a48cc3a2-3d38-47fb-84ab-66dec8e83f7a">

Reply:
@kotaaaa 

Thanks for your feedback!
As far as I know about responsive design, we don't have to care about screen sizes that do not exist as common sense.
The common narrowest mobile width is usually considered around 360px. Although I know that Galaxy Fold has a much narrower screen width of 280px.

These are the references: 
https://www.altamira.ai/blog/common-screen-sizes-for-responsive-web-design/
https://twinr.dev/blogs/a-complete-guide-to-android-screen-resolutions-and-sizes/

In 280px width, this Datepicker component still looks OK to me. What do you think?

Reply:
On second thought, I think I can take your advice because there might be a situation where it is used with another component in pairs, side by side, although that case does not exist in the current Figma design.

Reply:
Thank you for providing reference!! I felt both of your idea totally make sense so I will leave the decision to you!!

Reply:
@kotaaaa 
I fixed it as you suggested! I think it looks nicer with a narrow width.

### Before
<img width="360" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/f4346c05-9802-41c6-81a0-5bfac0c8b318">

### After
<img width="360" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/0f81eca4-8ee7-4077-bce3-6e437ddd469e">